dashboard/app/models/pd/workshop.rb

```diff
@@ -116,6 +116,10 @@ def self.attended_by(teacher)
     joins(sessions: :attendances).where(pd_attendances: {teacher_id: teacher.id}).distinct
   end
 
+  def self.find_by_section_code(section_code)
+    joins(:section).find_by(sections: {code: section_code})
```

Comment:
Yeah, in a class method. Same as [here](https://github.com/code-dot-org/code-dot-org/blob/2deb679ccbdce5152e6fa6979323e47996e5aa62/dashboard/app/models/pd/workshop.rb#L107-L117). It's equivalent to `Pd::Workshop.joins...`. We could convert these to [scopes](http://guides.rubyonrails.org/active_record_querying.html#scopes) if that's more clear.